plugins/woocommerce/includes/abstracts/abstract-wc-product.php

```diff
@@ -267,6 +267,9 @@ public function get_sku( $context = 'view' ) {
 	 * @return string price
 	 */
 	public function get_price( $context = 'view' ) {
+		if ( $this->is_on_sale() ) {
+			return $this->get_prop( 'sale_price', $context );
+		}
```

Comment:
This is a _pretty significant change_ to existing behaviour - especially considering there is an explicit `get_sale_price` method already available. The public contract for this method has now changed and I suspect could have a ripple effect across all code, referencing it for products that have a sale price. 

Please review and ensure this won't impact negatively existing usages.

Reply:
On the surface, this does look like the right fix, but looking at code [like this](https://github.com/woocommerce/woocommerce/blob/99aeaff6dd132e55520695d07af57322c0d7bc73/plugins/woocommerce/includes/wc-product-functions.php#L451) the price field for the product is _supposed_ to be updated to the sale price in the db. I'd be wary about shipping this change as it looks like it might just be a bandaid over the real problem (and might introduce problems of its own).

Reply:
To sum up my previous two comments:

- The code changed in this PR has been present for 7 years. Bugs can be in code that old, but `$product->get_price` is a pretty hot code path so we'd want to make sure any changes are not affecting the expected return value.
- The code added is an attempt to make sure `get_price` is returning the `sale price` (overriding what is in the db) if the product is on sale (to fix the apparent bug with scheduled sales not updating price).
- I'm concerned it's a bandaid over the actual problem, given the sale price is supposed to be [updated in the DB on schedule](https://github.com/woocommerce/woocommerce/blob/99aeaff6dd132e55520695d07af57322c0d7bc73/plugins/woocommerce/includes/wc-product-functions.php#L451).
- This fix on the surface looks like a good fix, but still changes the behaviour for the getter which may already be a bug accounted for in code in the wild. So the fix may have some unexpected side-effects if released as is (besides not addressing what may potentially be the real bug).

I'm concerned this fix is a shallow fix without giving appropriate thought to the ramifications of the change nor accounting for what the _real_ bug might be (for instance, why isn't the price updated in the DB?).

Reply:
We've created a https://github.com/woocommerce/woocommerce/pull/43094 as we are considering reverting the change.